sklearn/compose/tests/test_resampled.py

```diff
@@ -0,0 +1,60 @@
+from sklearn.base import BaseEstimator
+from sklearn.datasets import make_classification
+from sklearn.svm import SVC, OneClassSVM
+from sklearn.decomposition import PCA
+from sklearn.pipeline import Pipeline
+from sklearn.compose import ResampledTrainer
+from sklearn.utils.estimator_checks import check_estimator
+from sklearn.utils.validation import _num_samples
+
+
+class HalfSampler(BaseEstimator):
+    "Train with every odd sample"
+    def fit_resample(self, X, y, **kw):
+        if _num_samples(X) > 1 and getattr(X, 'format', None) not in ['dia',
+                                                                      'coo',
+                                                                      'bsr']:
+            return X[::2], y[::2]
+        return X, y
+
+
+def test_estimator_checks():
+    check_estimator(ResampledTrainer(HalfSampler(), SVC()))
+
+
```

Comment:
We should have tests that explicitly check that the predictor is fitted with the right data, in fit, fit_transform, fit_predict... and tests that check that those `fit_*` methods return data corresponding to the full data passed in.

Reply:
> and tests that check that those `fit_*` methods return data corresponding to the full data passed in.

I don't think I understand what you mean.